src/binder/bind/bind_reading_clause.cpp

```diff
@@ -137,86 +137,136 @@ std::unique_ptr<BoundReadingClause> Binder::bindLoadFrom(
+        switch (readerConfig.fileType) {
+        case FileType::CSV: {
+            validateNumColumns(columnTypes.size(), tmpColumnTypes.size());
+        }
+        case FileType::PARQUET: {
+            validateNumColumns(columnTypes.size(), tmpColumnTypes.size());
+            validateColumnTypes(columnNames, columnTypes, tmpColumnTypes);
+        } break;
+        case FileType::NPY: {
+            validateNumColumns(1, tmpColumnTypes.size());
+            columnNames.push_back(tmpColumnNames[0]);
+            columnTypes.push_back(tmpColumnTypes[0]->copy());
+        } break;
+        default:
+            break;
+        }
+    }
+}
+
+void Binder::sniffFile(const common::ReaderConfig& readerConfig, uint32_t fileIdx,
+    std::vector<std::string>& columnNames, std::vector<std::unique_ptr<LogicalType>>& columnTypes) {
+    switch (readerConfig.fileType) {
     case FileType::CSV: {
-        auto csvReader = SerialCSVReader(readerConfig->filePaths[0], *readerConfig);
+        auto csvReader = SerialCSVReader(readerConfig.filePaths[fileIdx], readerConfig);
         auto sniffedColumns = csvReader.sniffCSV();
         for (auto& [name, type] : sniffedColumns) {
-            detectedColumnNames.push_back(name);
-            detectedColumnTypes.push_back(type.copy());
+            columnNames.push_back(name);
+            columnTypes.push_back(type.copy());
         }
     } break;
     case FileType::PARQUET: {
-        auto reader = ParquetReader(readerConfig->filePaths[0], memoryManager);
+        auto reader = ParquetReader(readerConfig.filePaths[fileIdx], memoryManager);
         auto state = std::make_unique<processor::ParquetReaderScanState>();
         reader.initializeScan(*state, std::vector<uint64_t>{});
         for (auto i = 0u; i < reader.getNumColumns(); ++i) {
-            detectedColumnNames.push_back(reader.getColumnName(i));
-            detectedColumnTypes.push_back(reader.getColumnType(i)->copy());
+            columnNames.push_back(reader.getColumnName(i));
+            columnTypes.push_back(reader.getColumnType(i)->copy());
         }
     } break;
     case FileType::NPY: {
-        auto reader = NpyReader(readerConfig->filePaths[0]);
-        auto columnName = std::string("column0");
+        auto reader = NpyReader(readerConfig.filePaths[0]);
+        auto columnName = std::string("column" + std::to_string(fileIdx));
         auto columnType = bindFixedListType(reader.getShape(), reader.getType());
-        detectedColumnNames.push_back(columnName);
-        detectedColumnTypes.push_back(columnType->copy());
+        columnNames.push_back(columnName);
+        columnTypes.push_back(columnType->copy());
     } break;
     default:
         throw BinderException(StringUtils::string_format(
-            "Load from {} file is not supported.", FileTypeUtils::toString(fileType)));
+            "Cannot sniff header of file type {}", FileTypeUtils::toString(readerConfig.fileType)));
```

Comment:
We should add a test for this with RDF (turtle files?).

Reply:
Not now. I'll do after the RDF feature is complete.